### PR TITLE
Missing domain admin tests

### PR DESCRIPTION
**Pull request by jacekwegr**
This PR adds missing domain admin tests for GraphQL api

### CODE REVIEW

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/134296) / small_tests / c4082d9ddb87589a78ac2ddd6f0e2ec6114b16c9
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3745/134296/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3745/134296/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-09-02_13.07.32/lib.mongooseim.logs/run.2022-09-02_13.07.33/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/134962) / small_tests / c96b0058602693e4d86e3fa11bb99b7b642dd7ba
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3745/134962/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3745/134962/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-09-05_11.33.48/lib.mongooseim.logs/run.2022-09-05_11.33.49/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/134985) / small_tests / 687cd1e6b37a9b76264dbebeedad9a6c5229be32
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3745/134985/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3745/134985/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-09-05_12.47.28/lib.mongooseim.logs/run.2022-09-05_12.47.29/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/135067) / small_tests / fe0defa11f3d00e321203b76ec429b19947bc54c
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3745/135067/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3745/135067/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-09-06_07.32.14/lib.mongooseim.logs/run.2022-09-06_07.32.15/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/135245) / small_tests / d88944b83ee7b0e77e822cf54a9ab02e6bdd81b0
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3745/135245/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3745/135245/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-09-06_14.18.37/lib.mongooseim.logs/run.2022-09-06_14.18.38/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/135268) / small_tests / e3494a7790854f9ca618b30571c516af70630a43
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3745/135268/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3745/135268/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-09-06_15.02.18/lib.mongooseim.logs/run.2022-09-06_15.02.19/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/135316) / small_tests / ab5748e529008f377ca81790e7a962b0de5ccd14
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3745/135316/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3745/135316/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-09-07_07.37.21/lib.mongooseim.logs/run.2022-09-07_07.37.21/suite.log.html)

File: big_tests/tests/graphql_gdpr_SUITE.erl [hunk 1]
```diff
@@ -68,6 +77,16 @@ admin_gdpr_no_user_test(Config) ->
     Res = admin_retrieve_personal_data(<<"AAAA">>, domain(), <<"AAA">>, Config),
     ?assertEqual(<<"user_does_not_exist_error">>, get_err_code(Res)).
 
+domain_admin_retrive_user_data_no_permission(Config) ->
+    escalus:fresh_story_with_config(Config, [{alice_bis, 1}],
```
Comment (JanuszJakubiec) [inline big_tests/tests/graphql_gdpr_SUITE.erl:81]:
```suggestion
    escalus:fresh_story_with_config(Config, [{alice_bis, 1}],
```

File: big_tests/tests/graphql_helper.erl [hunk 2]
```diff
@@ -139,9 +141,8 @@ get_err_msg(Resp) ->
     get_err_msg(1, Resp).
 
 get_unauthorized({Code, #{<<"errors">> := Errors}}) ->
-    [#{<<"extensions">> := #{<<"code">> := ErrorCode}}] = Errors,
-    assert_response_code(unauthorized, Code),
-    ?assertEqual(<<"no_permissions">>, ErrorCode).
```
Comment (JanuszJakubiec) [inline big_tests/tests/graphql_helper.erl:144]: Why did you delete this assertion?

File: big_tests/tests/graphql_inbox_SUITE.erl [hunk 3]
```diff
@@ -146,6 +158,32 @@ admin_try_flush_nonexistent_host_type_bin(Config) ->
...
+    get_unauthorized(flush_domain_bin(<<"external-domain">>, Config)).
+
+domain_admin_flush_global_bin_no_permission(Config) ->
+    escalus:fresh_story_with_config(Config, [{alice, 1}, {alice_bis, 1}, {kate, 1}],
+                                    fun domain_admin_flush_global_bin_no_permission/4).
+
+domain_admin_flush_global_bin_no_permission(Config, Alice, AliceBis, Kate) ->
+    SecHostType = domain_helper:host_type(),
+    create_room_and_make_users_leave(Alice, AliceBis, Kate),
+    get_unauthorized(flush_global_bin(SecHostType, null, Config)). % check it
```
Comment (JanuszJakubiec) [inline big_tests/tests/graphql_inbox_SUITE.erl:182]: Why is there a comment?

File: big_tests/tests/graphql_last_SUITE.erl [hunk 4]
```diff
@@ -290,6 +333,61 @@ admin_logged_user_is_not_old_user_story(Config, _Alice) ->
...
+%% Domain admin old users test cases
+
+domain_admin_try_list_old_users_external_domain(Config) ->
+    get_unauthorized(admin_list_old_users(<<"external">>, now_dt_with_offset(0), Config)).
+
+domain_admin_try_remove_old_users_external_domain(Config) ->
+    get_unauthorized(admin_remove_old_users(<<"external">>, now_dt_with_offset(0), Config)).
+
+domain_admin_list_old_users_global(Config) ->
+    jids_with_config(Config, [alice, alice_bis, bob],
```
Comment (JanuszJakubiec) [inline big_tests/tests/graphql_last_SUITE.erl:356]:
```suggestion
    jids_with_config(Config, [alice, alice_bis, bob],
```

Comment (JanuszJakubiec) [inline big_tests/tests/graphql_last_SUITE.erl:356]:
```suggestion
    jids_with_config(Config, [alice, alice_bis, bob],
```

File: big_tests/tests/graphql_last_SUITE.erl [hunk 5]
```diff
@@ -290,6 +333,61 @@ admin_logged_user_is_not_old_user_story(Config, _Alice) ->
...
+    OldDateTime = now_dt_with_offset(100),
+
+    set_last(Bob, OldDateTime, Config),
+    set_last(AliceBis, OldDateTime, Config),
+    set_last(Alice, now_dt_with_offset(200), Config),
+
+    get_unauthorized(admin_list_old_users(null, now_dt_with_offset(150), Config)).
+
+domain_admin_remove_old_users_global(Config) ->
+    jids_with_config(Config, [alice, alice_bis, bob],
```
Comment (JanuszJakubiec) [inline big_tests/tests/graphql_last_SUITE.erl:369]:
```suggestion
    jids_with_config(Config, [alice, alice_bis, bob],
```

File: big_tests/tests/graphql_session_SUITE.erl [hunk 6]
```diff
@@ -54,11 +69,23 @@ init_per_group(admin_session_cli, Config) ->
     graphql_helper:init_admin_handler(Config);
 init_per_group(admin_session_http, Config) ->
     graphql_helper:init_admin_cli(Config);
+init_per_group(domain_admin_session, Config) ->
+    Config1 = graphql_helper:init_domain_admin_handler(Config),
+    case Config1 of
+        {skip, require_rdbms} ->
```
Comment (JanuszJakubiec) [inline big_tests/tests/graphql_session_SUITE.erl:75]:
```suggestion
        {skip, require_rdbms} ->
```

**Review by JanuszJakubiec — COMMENTED** [review]
Looks good, I've added some minor stylistic comments

File: big_tests/tests/graphql_helper.erl (hunk 2 above)
Comment (JanuszJakubiec) [reply big_tests/tests/graphql_helper.erl:144]: ok 👍🏻

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/135765) / small_tests / f84b41b1f8322dbe7efbc8c3eee762df87e2e96e
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3745/135765/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3745/135765/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-09-08_13.53.25/lib.mongooseim.logs/run.2022-09-08_13.53.25/suite.log.html)

**Review by JanuszJakubiec — APPROVED** [review]
Looks good

File: big_tests/tests/graphql_http_upload_SUITE.erl [hunk 7]
```diff
@@ -194,6 +221,10 @@ admin_http_upload_not_configured(Config) ->
     ?assertEqual(<<"module_not_loaded_error">>, get_err_code(Result)),
     ?assertEqual(<<"mod_http_upload is not loaded for this host">>, get_err_msg(Result)).
 
+domain_admin_get_url_no_permission(Config) ->
+    Result = admin_get_url(<<"AAAAA">>, <<"test">>, 123, <<"Test">>, 123, Config),
```
Comment (chrzaszcz) [inline big_tests/tests/graphql_http_upload_SUITE.erl:225]: I think it might be more valuable to test the secondary domain instead, because this way we would be sure that the domain admin can't access a different domain even if it exists - which is actually the main reason to have the domain admin at all.

File: big_tests/tests/graphql_last_SUITE.erl [hunk 8]
```diff
@@ -290,6 +333,61 @@ admin_logged_user_is_not_old_user_story(Config, _Alice) ->
...
+%% Domain admin test cases
+
+domain_admin_try_set_nonexistent_user_last(Config) ->
+    get_unauthorized(admin_set_last(?NONEXISTENT_JID, <<"status">>, null, Config)).
+
+domain_admin_get_nonexistent_user_last(Config) ->
+    get_unauthorized(admin_get_last(?NONEXISTENT_JID, Config)).
+
+domain_admin_try_count_external_domain_active_users(Config) ->
+    get_unauthorized(admin_count_active_users(<<"external-domain.com">>, null, Config)).
```
Comment (chrzaszcz) [inline big_tests/tests/graphql_last_SUITE.erl:345]: It would be nice to test an existing domain, e.g. `domain_helper:secondary_domain()` - this way we would be sure, that the domain admin is unauthorized to access an existing domain as well.

File: big_tests/tests/graphql_muc_light_SUITE.erl [hunk 9]
```diff
@@ -537,6 +559,214 @@ user_blocking_list_story(Config, Alice, Bob) ->
                     <<"entity">> => RoomBin}],
                  get_ok_value(?GET_BLOCKING_LIST_PATH, Res5)).
 
+%% Domain admin test cases
```
Comment (chrzaszcz) [inline big_tests/tests/graphql_muc_light_SUITE.erl:562]: A lot of tests are almost identical to the ones for the global admin. Could you reuse code instead? The story function could just have one more argument - the room name, as I think this is the only thing that is different.

File: big_tests/tests/graphql_token_SUITE.erl [hunk 10]
```diff
@@ -106,6 +117,14 @@ user_revoke_token_test(Config, Alice) ->
     ParsedRes = get_ok_value([data, token, revokeToken], Res2),
     ?assertEqual(<<"Revoked.">>, ParsedRes).
 
+% Domain admin tests
+
+domain_admin_request_token_no_user_test(Config) ->
+    get_unauthorized(admin_request_token(<<"AAAAA">>, Config)).
```
Comment (chrzaszcz) [inline big_tests/tests/graphql_token_SUITE.erl:123]: I think we should test real users at the secondary domain here.

File: big_tests/tests/graphql_vcard_SUITE.erl [hunk 11]
```diff
@@ -141,6 +156,32 @@ user_get_others_vcard_no_user(Config, Alice) ->
...
+
+domain_admin_get_vcard_no_user(Config) ->
+    get_unauthorized(get_vcard(<<"AAAAA">>, Config)).
+
+domain_admin_get_vcard_no_permission(Config) ->
+    escalus:fresh_story_with_config(Config, [{alice_bis, 1}],
+                                    fun domain_admin_get_vcard_no_permission/2).
+
+domain_admin_get_vcard_no_permission(Config, AliceBis) ->
+    Result = get_vcard(user_to_bin(AliceBis), Config),
```
Comment (chrzaszcz) [inline big_tests/tests/graphql_vcard_SUITE.erl:173]: This is what I meant in the comments above - test a user at the secondary domain.

File: big_tests/tests/graphql_offline_SUITE.erl [hunk 12]
```diff
@@ -135,6 +168,14 @@ admin_delete_old_messages_offline_not_configured_test(Config) ->
     Result = delete_old_messages(domain(), 2, Config),
     ?assertEqual(<<"module_not_loaded_error">>, get_err_code(Result)).
 
+%% Domain admin test cases
+
+domain_admin_delete_expired_messages_no_domain_test(Config) ->
+    get_unauthorized(delete_expired_messages(<<"AAAA">>, Config)).
```
Comment (chrzaszcz) [inline big_tests/tests/graphql_offline_SUITE.erl:174]: Could you test the secondary domain?

File: big_tests/tests/graphql_private_SUITE.erl [hunk 13]
```diff
@@ -143,6 +155,17 @@ private_input() ->
            attrs = [{<<"xmlns">>, "alice:private:ns"}],
            children = [{xmlcdata, <<"DATA">>}]}.
 
+%% Domain admin tests
+
+domain_admin_no_user_error_set(Config) ->
+    ElemStr = exml:to_binary(private_input()),
+    Result = admin_set_private(<<"AAAAA">>, ElemStr, Config),
```
Comment (chrzaszcz) [inline big_tests/tests/graphql_private_SUITE.erl:162]: Could you test a user at the secondary domain?

**Review by chrzaszcz — COMMENTED** [review]
It is going in the right direction in general. However, I think that in many cases the tests are not in line with the main goal - to check that the domain admin can't access a different (existing) domain - which is the main purpose of the domain permission level.

So, on one hand I see that we are repeating tests for non-existent domains (which is good, but I don't think it's necessary), but on the other hand sometimes we miss the most important test: for the secondary domain - either with `domain_helper:secondary_domain()`, or with a user like AliceBis. I didn't mark all such places, because some tests (e.g. `muc` and `muc_light`) are missing such tests entirely.

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/137322) / small_tests / cc3202fe3de9b4f90b97a036dc2a868219d0b22c
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3745/137322/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3745/137322/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-09-16_14.40.12/lib.mongooseim.logs/run.2022-09-16_14.40.13/suite.log.html)

File: big_tests/tests/graphql_last_SUITE.erl [hunk 14]
```diff
@@ -57,6 +66,25 @@ admin_old_users_tests() ->
      admin_user_without_last_info_is_old_user,
      admin_logged_user_is_not_old_user].
 
+domain_admin_last_tests() ->
+    [admin_set_last,
+     domain_admin_set_user_last_no_permission,
+     admin_get_last,
+     domain_admin_set_user_last_no_persmission,
```
Comment (chrzaszcz) [inline big_tests/tests/graphql_last_SUITE.erl:73]: Double typo?
```suggestion
     domain_admin_get_user_last_no_permission,
```

File: big_tests/tests/graphql_muc_SUITE.erl [hunk 15]
```diff
@@ -582,6 +631,148 @@ admin_try_list_nonexistent_room_affiliations(Config) ->
...
+domain_admin_try_delete_room_with_nonexistent_domain(Config) ->
+    RoomJID = jid:make_bare(<<"unknown">>, <<"unknown">>),
+    get_unauthorized(delete_room(RoomJID, null, Config)).
+
+domain_admin_create_and_delete_room_no_permission(Config) ->
+    Name = rand_name(),
+    ExternalDomain = domain_helper:secondary_domain(),
+    UnknownDomain = <<"unknown">>,
+    MUCServer = muc_helper:muc_host(),
+    % Create instant room with an non-existent domain
```
Comment (chrzaszcz) [inline big_tests/tests/graphql_muc_SUITE.erl:645]:
```suggestion
    % Create instant room with a non-existent domain
```

File: big_tests/tests/graphql_muc_SUITE.erl [hunk 16]
```diff
@@ -582,6 +631,148 @@ admin_try_list_nonexistent_room_affiliations(Config) ->
...
+    UnknownJID = <<(rand_name())/binary, "@", UnknownDomain/binary>>,
+    Res = create_instant_room(MUCServer, Name, UnknownJID, <<"Ali">>, Config),
+    get_unauthorized(Res),
+    % Create instant room with an external domain
+    ExternalJID = <<(rand_name())/binary, "@", ExternalDomain/binary>>,
+    Res2 = create_instant_room(MUCServer, Name, ExternalJID, <<"Ali">>, Config),
+    get_unauthorized(Res2),
+    % Delete instant room with an non-existent domain
```
Comment (chrzaszcz) [inline big_tests/tests/graphql_muc_SUITE.erl:653]:
```suggestion
    % Delete instant room with a non-existent domain
```

File: big_tests/tests/graphql_muc_light_SUITE.erl [hunk 17]
```diff
@@ -537,6 +580,229 @@ user_blocking_list_story(Config, Alice, Bob) ->
...
+    AliceBin = escalus_client:short_jid(Alice),
+    BobBin = escalus_client:short_jid(Bob),
+    MUCServer = ?config(muc_light_host, Config),
+    ExternalDomain = domain_helper:secondary_domain(),
+    RoomName = <<"first room">>,
+    {ok, #{jid := #jid{luser = RoomID} = RoomJID}} =
+        create_room(MUCServer, RoomName, <<"subject">>, AliceBin),
+    {ok, _} = invite_user(RoomJID, AliceBin, BobBin),
+    % Try to change the config with a non-existent domain
+    Res = change_room_configuration(
```
Comment (chrzaszcz) [inline big_tests/tests/graphql_muc_light_SUITE.erl:636]: I have to note that both jid's in this test point to non-existing rooms at domains that are not MUC-Light domains. It would be better to test an existing room at a different domain.

File: big_tests/tests/graphql_muc_light_SUITE.erl [hunk 18]
```diff
@@ -537,6 +580,229 @@ user_blocking_list_story(Config, Alice, Bob) ->
...
+
+domain_admin_delete_room_no_permission_story(Config, Alice) ->
+    AliceBin = escalus_client:short_jid(Alice),
+    Name = <<"first room">>,
+    MUCServer = ?config(muc_light_host, Config),
+    ExternalDomain = domain_helper:secondary_domain(),
+    {ok, #{jid := #jid{luser = RoomID}}} =
+        create_room(MUCServer, Name, <<"subject">>, AliceBin),
+    % Try with a non-existent domain
+    Res = delete_room(make_bare_jid(RoomID, ?UNKNOWN_DOMAIN), Config),
```
Comment (chrzaszcz) [inline big_tests/tests/graphql_muc_light_SUITE.erl:669]: I have to note that both jid's in this test point to non-existing rooms at domains that are not MUC-Light domains. It would be better to test an existing room at a different domain.

File: big_tests/tests/graphql_muc_light_SUITE.erl [hunk 19]
```diff
@@ -537,6 +580,229 @@ user_blocking_list_story(Config, Alice, Bob) ->
...
+    ExternalDomain = domain_helper:secondary_domain(),
+    BobBin = escalus_client:short_jid(Bob),
+    MUCServer = ?config(muc_light_host, Config),
+    Name = <<"first room">>,
+    {ok, #{jid := #jid{luser = RoomID} = RoomJID}} =
+        create_room(MUCServer, Name, <<"subject">>, AliceBisBin),
+    {ok, _} = invite_user(RoomJID, AliceBisBin, BobBin),
+    ?assertEqual(2, length(get_room_aff(RoomJID))),
+    % Try with a non-existent domain
+    Res = kick_user(make_bare_jid(RoomID, ?UNKNOWN_DOMAIN), BobBin, Config),
```
Comment (chrzaszcz) [inline big_tests/tests/graphql_muc_light_SUITE.erl:690]: I have to note that both jid's in this test point to non-existing rooms at domains that are not MUC-Light domains. It would be better to test an existing room at a different domain.

File: big_tests/tests/graphql_muc_light_SUITE.erl [hunk 20]
```diff
@@ -537,6 +580,229 @@ user_blocking_list_story(Config, Alice, Bob) ->
...
+domain_admin_get_room_messages_no_permission_story(Config, Alice) ->
+    AliceBin = escalus_client:short_jid(Alice),
+    MUCServer = ?config(muc_light_host, Config),
+    RoomName = <<"first room">>,
+    ExternalDomain = domain_helper:secondary_domain(),
+    Limit = 40,
+    {ok, #{jid := #jid{luser = RoomID}}} =
+    create_room(MUCServer, RoomName, <<"subject">>, AliceBin),
+    % Try with a non-existent domain
+    Res = get_room_messages(make_bare_jid(RoomID, ?UNKNOWN_DOMAIN), Limit, null, Config),
```
Comment (chrzaszcz) [inline big_tests/tests/graphql_muc_light_SUITE.erl:729]: I have to note that both jid's in this test point to non-existing rooms at domains that are not MUC-Light domains. It would be better to test an existing room at a different domain.

File: big_tests/tests/graphql_muc_light_SUITE.erl [hunk 21]
```diff
@@ -537,6 +580,229 @@ user_blocking_list_story(Config, Alice, Bob) ->
...
+
+domain_admin_list_user_rooms_no_permission_story(Config, Alice) ->
+    AliceBin = escalus_client:short_jid(Alice),
+    MUCServer = ?config(muc_light_host, Config),
+    RoomName = <<"first room">>,
+    ExternalDomain = domain_helper:secondary_domain(),
+    {ok, #{jid := #jid{luser = RoomID}}} =
+    create_room(MUCServer, RoomName, <<"subject">>, AliceBin),
+    % Try with a non-existent domain
+    Res = list_user_rooms(make_bare_jid(RoomID, ?UNKNOWN_DOMAIN), Config),
```
Comment (chrzaszcz) [inline big_tests/tests/graphql_muc_light_SUITE.erl:747]: This does not make much sense to me: it was supposed to list rooms for a user, so we need to pass a user JID, not a room JID.

File: big_tests/tests/graphql_muc_light_SUITE.erl [hunk 22]
```diff
@@ -537,6 +580,229 @@ user_blocking_list_story(Config, Alice, Bob) ->
...
+                                    fun domain_admin_list_room_users_story_no_permission/2).
+
+domain_admin_list_room_users_story_no_permission(Config, Alice) ->
+    AliceBin = escalus_client:short_jid(Alice),
+    MUCServer = ?config(muc_light_host, Config),
+    RoomName = <<"first room">>,
+    ExternalDomain = domain_helper:secondary_domain(),
+    {ok, #{jid := #jid{luser = RoomID}}} = create_room(MUCServer, RoomName, <<"subject">>, AliceBin),
+    % Try with a non-existent domain
+    Res = list_room_users(make_bare_jid(RoomID, ?UNKNOWN_DOMAIN), Config),
```
Comment (chrzaszcz) [inline big_tests/tests/graphql_muc_light_SUITE.erl:764]: I have to note that both jid's in this test point to non-existing rooms at domains that are not MUC-Light domains. It would be better to test an existing room at a different domain.

File: big_tests/tests/graphql_token_SUITE.erl [hunk 23]
```diff
@@ -106,6 +117,31 @@ user_revoke_token_test(Config, Alice) ->
...
+
+domain_admin_request_token_no_permission_test(Config) ->
+    escalus:fresh_story_with_config(Config, [{alice_bis, 1}],
+                                    fun domain_admin_request_token_no_permission_test/2).
+
+domain_admin_request_token_no_permission_test(Config, AliceBis) ->
+    % External domain user
+    Res = admin_request_token(user_to_bin(AliceBis), Config),
+    get_unauthorized(Res),
+    % Non-exisitng user
```
Comment (chrzaszcz) [inline big_tests/tests/graphql_token_SUITE.erl:130]:
```suggestion
    % Non-existing user
```

File: big_tests/tests/graphql_token_SUITE.erl [hunk 24]
```diff
@@ -106,6 +117,31 @@ user_revoke_token_test(Config, Alice) ->
...
+    get_unauthorized(Res2).
+
+domain_admin_revoke_token_no_permission_test(Config) ->
+    escalus:fresh_story_with_config(Config, [{alice_bis, 1}],
+                                    fun domain_admin_revoke_token_no_permission_test/2).
+domain_admin_revoke_token_no_permission_test(Config, AliceBis) ->
+    % External domain user
+    Res = admin_revoke_token(user_to_bin(AliceBis), Config),
+    get_unauthorized(Res),
+    % Non-exisitng user
```
Comment (chrzaszcz) [inline big_tests/tests/graphql_token_SUITE.erl:141]:
```suggestion
    % Non-existing user
```

File: big_tests/tests/graphql_offline_SUITE.erl [hunk 25]
```diff
@@ -61,18 +82,30 @@ init_per_group(admin_http, Config) ->
...
     Backend = mongoose_helper:get_backend_mnesia_rdbms_riak(HostType),
     ModConfig = create_config(Backend),
     dynamic_modules:ensure_modules(HostType, ModConfig),
     [{backend, Backend} | escalus:init_per_suite(Config)];
+init_per_group(domain_admin_offline, Config) ->
+    HostType = host_type(),
+    Backend = mongoose_helper:get_backend_mnesia_rdbms_riak(HostType),
+    ModConfig = create_config(Backend),
+    dynamic_modules:ensure_modules(HostType, ModConfig),
+    [{backend, Backend} | escalus:init_per_suite(Config)];
```
Comment (chrzaszcz) [inline big_tests/tests/graphql_offline_SUITE.erl:98]: For me it's too much repetition
```suggestion
init_per_group(GroupName, Config) when GroupName =:= admin_offline;
                                       GroupName =:= domain_admin_offline ->
    HostType = host_type(),
    Backend = mongoose_helper:get_backend_mnesia_rdbms_riak(HostType),
    ModConfig = create_config(Backend),
    dynamic_modules:ensure_modules(HostType, ModConfig),
    [{backend, Backend} | escalus:init_per_suite(Config)].
```

File: big_tests/tests/graphql_private_SUITE.erl [hunk 26]
```diff
@@ -143,6 +155,30 @@ private_input() ->
            attrs = [{<<"xmlns">>, "alice:private:ns"}],
            children = [{xmlcdata, <<"DATA">>}]}.
 
+%% Domain admin tests
+
+
```
Comment (chrzaszcz) [inline big_tests/tests/graphql_private_SUITE.erl:160]:
```suggestion

```

File: big_tests/tests/graphql_stanza_SUITE.erl [hunk 27]
```diff
@@ -40,6 +43,26 @@ admin_get_last_messages_cases() ->
...
+
+domain_admin_get_last_messages_cases() ->
+    [admin_get_last_messages,
+     admin_get_last_messages_for_unknown_user,
+     admin_get_last_messages_with,
+     admin_get_last_messages_limit,
+     admin_get_last_messages_limit_enforced,
+     admin_get_last_messages_before,
+     domain_admin_get_last_messages_no_permission].
+
```
Comment (chrzaszcz) [inline big_tests/tests/graphql_stanza_SUITE.erl:65]:
```suggestion
     domain_admin_send_stanza_from_unknown_domain,
     domain_admin_get_last_messages_no_permission]
    ++ admin_get_last_messages_cases().
```

**Review by chrzaszcz — COMMENTED** [review]
A lot of good tests here. I added some comments, mostly minor ones.

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/137627) / small_tests / 82aa71fcfa04165436e67f7c3e7fad0ab392ad55
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3745/137627/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3745/137627/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-09-20_08.24.25/lib.mongooseim.logs/run.2022-09-20_08.24.26/suite.log.html)

**Review by chrzaszcz — APPROVED** [review]
All looks good now 👌
